src/jit/emit.cpp

```diff
@@ -3048,11 +3056,12 @@ emitter::instrDesc *emitter::emitNewInstrCallDir(int        argCnt,
      */
     bool gcRefRegsInScratch = ((gcrefRegs & RBM_CALLEE_TRASH) != 0);
 
-    if  (!VarSetOps::IsEmpty(emitComp, GCvars)     ||   // any frame GCvars live
-         gcRefRegsInScratch          ||   // any register gc refs live in scratch regs
-         (byrefRegs != 0)            ||   // any register byrefs live
-         (argCnt > ID_MAX_SMALL_CNS) ||   // too many args
-         (argCnt < 0)                   ) // caller pops arguments
+    if  (!VarSetOps::IsEmpty(emitComp, GCvars)  ||                              // any frame GCvars live
+         gcRefRegsInScratch                     ||                              // any register gc refs live in scratch regs
+         (byrefRegs != 0)                       ||                              // any register byrefs live
+         (argCnt > ID_MAX_SMALL_CNS)            ||                              // too many args
+         (argCnt < 0)                                                           // caller pops arguments
+         FEATURE_UNIX_AMD64_STRUCT_PASSING_ONLY(|| (retSize1In != EA_UNKNOWN))) // There is a second ref/byref return register.
```

Comment:
The check is needed to make the instCall a LargeCall. This is necessary to store GC type for the second return register (if it is of GCRef type). Note that the first register GC type is stored in the base (small instruction) and it seems to apply for every instruction. The second return register is only applicable for calls. Adding it to the base instruction would be a waste in most of the cases - the size of the base instruction struct is very carefully maintained to be 128 bits (there is no more space to add new flags).
Adding the extra return register GC type to the LargeCall instruction descriptor made sense to me. We just need to make sure we are creating such instruction descriptor if the second return register happen to be a GC type. An example for this would be:
struct S
{
    Foo foo1;
    Foo foo2;
}

class SomeClass
{
    S foobar()
    {
        ........
    }

    // Somewhere there is a call to foobar().
    // In such case a large call descriptor needs to be created since the second return register
    // GC type needs to be maintained.
}